hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -488,7 +502,10 @@ private boolean doConnectToCandidateCluster(CandidateClusterContext context, boo
                 for (Member member : memberList) {
                     checkClientActive();
                     triedAddressesPerAttempt.add(member.getAddress());
-                    Connection connection = connect(member, o -> getOrConnectToMember((Member) o, switchingToNextCluster));
+                    connectionProcessListener.attemptingToConnectToAddress(member.getAddress());
```

Comment:
Possibly, the `member.getAddress()` and configured address (you have used below) can point to different addresses, even though they are the same thing. Like in the following simple case,

```java
        Hazelcast.newHazelcastInstance();

        ClientConfig clientConfig = new ClientConfig();
        clientConfig.getNetworkConfig().addAddress("localhost:5701");
        HazelcastClient.newHazelcastClient(clientConfig);
```

address would be localhost:5701, whereas member#address would be something like `192.168.1.2:5701` or something like it. should we handle it somehow?

Reply:
I think both provide sufficient information for the caller.